Doc/glossary.rst

```diff
@@ -40,16 +40,18 @@ Glossary
       ABCs with the :mod:`abc` module.
 
    annotation
-      A metadata value associated with a global variable, a class attribute or a
-      function or method parameter or return value, that stores a
-      :term:`type hint`.
+      A label with no defined semantics associated with a variable, a class
+      attribute or a function or method parameter or return value.
```

Comment:
Please drop "or method" here and everywhere else that you wrote it -- a method is defined as a kind of function already, and the phrase "function or method parameter or return value" is too awkward.